app/models/generic_object.rb

```diff
@@ -84,4 +94,21 @@ def property_setter(name, value)
 
     self.properties = properties.merge(name => val)
   end
+
+  # Suppose the methods are located under /Generic/Methods
+  # go.my_host(:user_id=>u.id)
+  def call_automate(method_name, options = {}, q_options = {})
+    raise "user_id is not specified in method [#{method_name}]" unless options[:user_id]
+
+    options.reverse_merge!(
```

Comment:
Do you expect the caller provide `:namespace` and `:class_name` in the `options` and otherwise use the default? This makes me feel we should save the method path in the definition. So it is no longer an array, but a hash like property attributes and associations. Besides the method name does not need to be equal to `:instance_name`.